src/tools/sssctl/sssctl_data.c

```diff
@@ -311,3 +314,222 @@ errno_t sssctl_cache_expire(struct sss_cmdline *cmdline,
     talloc_free(args);
     return ret;
 }
+
+errno_t get_confdb_domains(TALLOC_CTX *ctx, struct confdb_ctx *confdb,
+                           char ***_domains)
+{
+    int ret;
+    int domain_count = 0;
+    int i = 0;
+    struct sss_domain_info *domain = NULL;
+    struct sss_domain_info *domain_list = NULL;
+    char **domains;
+
+    TALLOC_CTX *tmp_ctx = talloc_new(NULL);
+
+    if (tmp_ctx == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc_new() failed\n");
+        return ENOMEM;
+    }
+
+    /* get domains */
+    ret = confdb_get_domains(confdb, &domain_list);
+    if (ret != EOK)
+        DEBUG(SSSDBG_CRIT_FAILURE, "Unable to get domain list\n");
+
+    for (domain = domain_list;
+         domain;
+         domain = get_next_domain(domain, 0)) {
+        domain_count++;
+    }
+
+    /* allocate output space */
+    domains = talloc_array(ctx, char *, domain_count + 1);
+    if (domains == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Could not allocate memory for sections\n");
```

Comment:
Did you mean domains instead of sections?

Reply:
Yes.